designs/2024-support-ts-config-files/README.md

```diff
@@ -0,0 +1,130 @@
+- Repo: eslint/eslint
+- Start Date: 2024-03-09
+- RFC PR: <https://github.com/eslint/rfcs/pull/117>
+- Authors: [Arya Emami](https://github.com/aryaemami59)
+
+# Add Support for TypeScript Config Files
+
+## Summary
+
+Add support for TypeScript config files (`eslint.config.ts`, `eslint.config.mts`, `eslint.config.cts`)
+
+<!-- One-paragraph explanation of the feature. -->
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+The primary motivation for adding support for TypeScript configuration files to ESLint is to enhance the developer experience and accommodate the evolving JavaScript ecosystem. As TypeScript's popularity continues to grow, more projects are adopting TypeScript not only for their source code but also for their configuration files. This shift is driven by TypeScript's ability to provide compile-time type checks and IntelliSense. By supporting `eslint.config.ts`, `eslint.config.mts`, and `eslint.config.cts`, ESLint will offer first-class support to TypeScript users, allowing them to leverage these benefits directly within their ESLint configuration.
+
+## Detailed Design
```

Comment:
It would be worth providing an example of a basic TypeScript config file, including the type annotation of the exported object, to illustrate the nice syntax and type-checking it enables.

Reply:
That's a great idea. I will do that, thanks!